### ISSUE TITLE
关于损失函数

### ISSUE THREAD

**Issue by AndrewTal** [WeijiaZhang24/CausalMIL#2]
您好，请问以下代码中的损失部分和论文中的loss function对应关系对嘛？

reconstruction_loss = loss(x_recon,x_max)   对应 logpf(x∗|z∗)

auxiliary_loss_y= loss(y_hat.squeeze(), reordered_y) 对应 αlogpω(Y|z∗)

KL_loss =  0.5 * (prior_mu.pow(2) + prior_std.pow(2) - 2*torch.log(prior_std) - 1).sum(dim=-1).mean() 对应 KL[qψ(z|B)||p(B)]

lpdf_qz_xu = -0.5 * ( 2*instance_std_hat.log() + (ins_hat - instance_mu_hat).pow(2).div(instance_std_hat.pow(2)))
lpdf_pzu = -0.5 * (  2*prior_std.log() + ( ins_hat- prior_mu).pow(2).div(prior_std.pow(2)))
(lpdf_qz_xu.sum(dim=-1) - lpdf_pzu.sum(dim=-1)).mean() 上面三行对应 KL[qφ(z∗|x∗,B)||pT,λ(z∗|B)]

另外 log pθ(B|z) 对应的损失项目应该是？

**Comment by WeijiaZhang24** [comment]
可能是发现 log pθ(B|z) 对结果的影响不大，所以该项移除了。不过时间有点久了，不是100%确定。

**Comment by AndrewTal** [comment]
got it
